src/covariances.jl

```diff
@@ -0,0 +1,80 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENCE in the project root.
+# ------------------------------------------------------------------
+
+using Statistics
+
+
+"""
+    designmatrix(comps)
+    
+Converts a vector of Composition objects into a Matrix of numbers.
+"""
+function designmatrix(comps)
+    matrix = reduce(hcat, components.(comps)[:,1])'
+    return matrix
+end
+
+"""
+    compvarmatrix(comps)
+
+Returns the compositional variation array, from definition 4.3 of Aitchson - The Statistical Analysis of Compositional Data.
+"""
+function compvarmatrix(comps)
+    dm = designmatrix(comps)
```

Comment:
Let's adopt the notation X for the design matrix.

Reply:
Done in commit aa2898e0c4dfdf5616f7d24604cfed53f0f56efc, but review is needed.